### ISSUE TITLE
Unknown collation: 'utf8mb4_0900_ai_ci' for Laravel 11 database

### ISSUE THREAD

**Issue by pa-purpura** [ddev/ddev#5962]
### TL;DR

DDEV will have explicit support for Laravel 11 with the next major release v1.23.0

For DDEV v1.22.7, use the quickstart below:

```bash
mkdir my-laravel-app
cd my-laravel-app
ddev config --project-type=laravel --docroot=public --create-docroot --php-version=8.2
ddev composer create --prefer-dist --no-install --no-scripts laravel/laravel -y
ddev composer install
ddev exec "php artisan key:generate"
ddev exec "sed -i 's/^MAIL_MAILER=.*/MAIL_MAILER=smtp/' .env"
ddev exec "echo DB_COLLATION=utf8mb4_unicode_ci >> .env && php artisan migrate"
ddev launch
```

If you want to use SQLite database for Laravel 11, use this `ddev config` instead:

```
ddev config --project-type=laravel --docroot=public --create-docroot --php-version=8.2 --omit-containers=db --disable-settings-management=true
```

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Output of `ddev debug test`

<details><summary>Expand `ddev debug test` diagnostic information</summary>

```
[COPY-PASTE HERE THE OUTPUT OF `ddev debug test`]
```
</details>         

### Expected Behavior

After a " ddev launch" i exepect a tipical laravel welcome page

### Actual Behavior

i land on an error page, with this message 
```
Illuminate\Database\QueryException
PHP 8.2.9
11.0.0
SQLSTATE[HY000]: General error: 1273 Unknown collation: 'utf8mb4_0900_ai_ci' 
```

### Anything else?

First of all, i changed the version of php:
`ddev config --project-type=laravel --docroot=public --create-docroot --php-version=8.1`
to `... --php-version=8.2`, otherwise by default Laravel install version 10.4.
Then i got SQLSTATE error.
I think the problem is due to new version of Laravel that has SQLite connection to db, by default, i tried to changed manually `.env` file, but it didn't work.

**Comment by rfay** [comment]
Hi @pa-purpura - I imagine you are having a problem that requires MySQL 8.0?

That collation is available only in mysql.

`ddev delete -Oy` and `ddev config --database=mysql:8.0` and then `ddev start` and import your database again?

**Comment by stasadev** [comment]
In Laravel 11, the default collation has been changed, as an alternative, add this to `.env` (the default value from Laravel 10):

```dotenv
DB_COLLATION=utf8mb4_unicode_ci
```

---

I've prepared a PR to support Laravel 11, test artifacts are available [here](https://github.com/ddev/ddev/pull/5963#issuecomment-1992641146).

**Comment by tyler36** [comment]
I didn't encounter this when using https://github.com/ddev/ddev/issues/5962 and a fresh Laravel install.
I was able to access the welcome page and run `migration:fresh` with seeder.

I suspect this might be an issue when upgrading an existing project?

**Comment by stasadev** [comment]
> I suspect this might be an issue when upgrading an existing project?
No, this is an error with fresh Laravel install using https://ddev.readthedocs.io/en/stable/users/quickstart/#laravel with DDEV v1.22.7

I updated the OP with a "fix" for our stable.

**Comment by pa-purpura** [comment]
Thank you  @stasadev i have ddev version 1.22.3, i followed the quickstart that you advise, and everything works fine.
Thanks!

**Comment by stasadev** [comment]
I updated the OP, the workaround is no longer needed if you are using Laravel 11.0.3.
